lib/core.js

```diff
@@ -131,6 +131,9 @@ module.exports = class Core {
     header.tree.rootHash = unslab(header.tree.rootHash)
     header.tree.signature = unslab(header.tree.signature)
 
+    if (header.keyPair?.publicKey) header.keyPair.publicKey = unslab(header.keyPair.publicKey)
+    if (header.keyPair?.secretKey) header.keyPair.secretKey = unslab(header.keyPair.secretKey)
```

Comment:
if (header.keyPair) {
  header.keyPair.publicKey = unslab(...)
  header.keyPair.secretKey = unslab(...)
}

Reply:
That was my initial approach, but that doesn't work for the case where keyPair only contains a publicKey, and no secretKey

Reply:
publicKey is not null?

Reply:
if thats undefined and on the header, thats a bug

Reply:
Ok, there's a test case which hits that, so that's a bug, looking into it

Reply:
Solved by setting secretKey to null here is it's not set: https://github.com/holepunchto/hypercore/pull/551/files#diff-826e5ed9069c92baa50ec46dde8074a6789b4c90837c5ede0e6b0438e3c34309R124-R126